### DISCUSSION TITLE
primary star mass

### DISCUSSION THREAD

**Discussion by Karytza** [phoebe-project/phoebe2#568] [Questions]
How can I change the mass of my primary star?
When I added the components manually I declared the value of the mass of my primary star, but later it was changed to another value and now I don't know how to change it back. I read that mass values ​​were restricted to Kepler's Laws, but this is not clear to me yet. I did this:
b = phoebe.Bundle()
b.add_component(phoebe.component.star, component='primary', mass=1.2)
b.add_component('star', component='secondary')

**Comment by kecnry** [comment]
Unless you really need to create a custom system, I'd suggest starting with the default binary (`b = phoebe.default_binary()` which already has the orbit attached to two stars).  At that point you should be able to see the Kepler's third law [constraint](http://phoebe-project.org/docs/latest/tutorials/constraints) and can "flip" it so that the mass is adjustable.  You'll do something like: `b.flip_constraint('mass@primary', solve_for='period')` where you can choose what other parameter will be read-only instead (and therefore not adjustable) via the `solve_for` argument.  Depending on your science case, you'll likely either want period or semi-major axis (`sma`).

If you want to build the system manually, you'll still need to add an orbit and a hierarchy.  Unfortunately there isn't currently any way to dictate in advance the "directions" of the constraints, so any mass you provide will be overwritten based on the period, mass-ratio, and semi-major axis of the orbit so you would have to do the same as above or do the math in advance to make sure those values are set correctly.

Hope this helps!  Let us know if you run into any issues!

-Kyle

**Reply by Karytza** [reply]
The situation is that I need to model a binary system to calculate the mass of a secondary star, I have the radial velocities of my primary star, the mass of the primary star, the orbital period and the mean eclipse time. That said, is it better to use the default system? My thought was that it was necessary to create a system manually, but I am really still learning how to use Phoebe.

**Reply by kecnry** [reply]
Yes, I'd almost always suggest starting with the default system and modifying the parameter values and constraints from there rather than building from scratch.  It sounds like you're trying to get a mass ratio (and through that and a known/assumed primary mass, compute the secondary mass) from the light curve? Just be aware that that is poorly constrained and really only possible in a few scenarios (you'll likely run into a lot of parameter degenracies).
